python/ray/tests/test_failure.py

```diff
@@ -724,29 +741,49 @@ def sleep_to_kill_raylet():
     thread.join()
 
 
-def test_connect_with_disconnected_node(shutdown_only):
-    config = json.dumps({
-        "num_heartbeats_timeout": 50,
-        "heartbeat_timeout_milliseconds": 10,
-    })
-    cluster = Cluster()
-    cluster.add_node(num_cpus=0, _internal_config=config)
-    ray.init(redis_address=cluster.redis_address)
+def test_connect_with_disconnected_node(short_heartbeat_timeout):
+    cluster, _ = short_heartbeat_timeout
+    cluster.add_node(num_cpus=0)
     info = relevant_errors(ray_constants.REMOVED_NODE_ERROR)
     assert len(info) == 0
     # This node is killed by SIGKILL, ray_monitor will mark it to dead.
-    dead_node = cluster.add_node(num_cpus=0, _internal_config=config)
+    dead_node = cluster.add_node(num_cpus=0)
     cluster.remove_node(dead_node, allow_graceful=False)
     wait_for_errors(ray_constants.REMOVED_NODE_ERROR, 1, timeout=2)
     # This node is killed by SIGKILL, ray_monitor will mark it to dead.
-    dead_node = cluster.add_node(num_cpus=0, _internal_config=config)
+    dead_node = cluster.add_node(num_cpus=0)
     cluster.remove_node(dead_node, allow_graceful=False)
     wait_for_errors(ray_constants.REMOVED_NODE_ERROR, 2, timeout=2)
     # This node is killed by SIGTERM, ray_monitor will not mark it again.
-    removing_node = cluster.add_node(num_cpus=0, _internal_config=config)
+    removing_node = cluster.add_node(num_cpus=0)
     cluster.remove_node(removing_node, allow_graceful=True)
     with pytest.raises(Exception, match=('Timing out of wait.')):
         wait_for_errors(ray_constants.REMOVED_NODE_ERROR, 3, timeout=2)
     # There is no connection error to a dead node.
     info = relevant_errors(ray_constants.RAYLET_CONNECTION_ERROR)
     assert len(info) == 0
+
+
+def test_init_exception_in_checkpointable_actor(short_heartbeat_timeout):
```

Comment:
maybe put this test in `test_actor.py`, because other checkpoint tests are there.

Reply:
Yes. I put this test in `test_actor.py` at the beginning, but `wait_for_errors` only in `test_failure.py`. Maybe later on we need to refactor to test code to put some useful functions together and remove the duplicated pytest fixtures.
I will move this function to `test_actor.py` and submit another PR to refactor the pytest fixtures.